File: codeforces/cf1542B_plus_and_multiply.py
# uninhm
# https://codeforces.com/contest/1542/problem/B
# modulo, greedy

for _ in range(int(input())):
    n, a, b = map(int, input().split())
    m = n%b
    if a == 1:
        if (n-1)%b == 0:
            print("Yes")
        else:
            print("No")
        continue
    x = 1
    while x <= n:
        if (x%b) == m:
            print("Yes")
            break
        x *= a
    else:
        print("No")

File: codeforces/cf1542B_plus_and_multiply.c
// uninhm
// https://codeforces.com/contest/1542/problem/B
// modulo, greedy

#include <stdio.h>

int main ()
{
  long long t;
  scanf ("%lld\n", &t);
  while (t--)
    {
      long long n, a, b;
      scanf ("%lld %lld %lld\n", &n, &a, &b);
      long long m = n % b;

      if (a == 1)
        {
          if ((n-1)%b == 0)
            puts ("Yes");
          else
            puts ("No");
          continue;
        }

      long long x = 1;
      long long found = 0;
      while (x <= n)
        {
          if (x%b == m)
            {
              found = 1;
              break;
            }
          x *= a;
        }
      if (found)
        puts ("Yes");
      else
        puts ("No");
    }
}
